web/source/dom/targets/input.ts

```diff
@@ -55,13 +59,13 @@ namespace com.keyman.dom.targets {
 
     getCaret(): number {
       if(this.root.selectionStart == this._cachedSelectionStart) {
-        return this.processedSelectionStart;
+        return this.processedSelectionEnd;
       } else {
         this._cachedSelectionStart = this.root.selectionStart; // KMW-1
         this.processedSelectionStart = this.root.value._kmwCodeUnitToCodePoint(this.root.selectionStart); // I3319
         this.processedSelectionEnd = this.root.value._kmwCodeUnitToCodePoint(this.root.selectionEnd);     // I3319
 
-        return this.processedSelectionStart;
+        return this.processedSelectionEnd;
       }
```

Comment:
To me this code would be easier to understand like this:

```suggestion
      if (this.root.selectionStart != this._cachedSelectionStart) {
        this._cachedSelectionStart = this.root.selectionStart; // KMW-1
        this.processedSelectionStart = this.root.value._kmwCodeUnitToCodePoint(this.root.selectionStart); // I3319
        this.processedSelectionEnd = this.root.value._kmwCodeUnitToCodePoint(this.root.selectionEnd);     // I3319
      }
      return this.processedSelectionEnd;
```